site/documentation/content/admin-guide/hono-kafka-client-configuration.md

```diff
@@ -0,0 +1,49 @@
++++
+title = "Hono Kafka Client Configuration"
+weight = 342
++++
+
+Protocol adapters can be configured to use Kafka for the messaging. The Kafka client used there can be configured with 
+environment variables and/or command line options.
+
+{{% note title="Tech preview" %}}
+The support of Kafka as a messaging system is currently a preview and not yet ready for production. 
+The implementation as well as it's APIs may change with the next version. 
+{{% /note %}}
+
+## Producer Configuration Properties
+
+The `org.eclipse.hono.kafka.client.CachingKafkaProducerFactory` factory can be used to create Kafka producers for Hono's Kafka based APIs. 
+The producers created by the factory are configured with instances of the class `org.eclipse.hono.kafka.client.KafkaProducerConfigProperties`
+which can be used to programmatically configure a producer. 
+
+The configuration needs to be provided in the form `HONO_KAFKA_PRODUCERCONFIG_${PROPERTY}` as an environment variable or
+as a command line option in the form `hono.kafka.producerConfig.${property}`, where `${PROPERTY}` respectively 
+`${property}` is any of the Kafka client's [producer properties](https://kafka.apache.org/documentation/#producerconfigs). 
+The provided configuration is passed directly to the Kafka producer without Hono parsing or validating it.
+The following table shows which properties _are_ changed by Hono.
+
+| Environment Variable<br>Command Line Option | Mandatory Value | Description |
+| :------------------------------------------ | :-------------- | :---------- |
+| `HONO_KAFKA_PRODUCERCONFIG_KEY_SERIALIZER`<br>`--hono.kafka.producerConfig.key.serializer` | `org.apache.kafka.common.serialization.StringSerializer` | The record keys in Hono are always strings. Any other specified value is ignored. |
```

Comment:
why do we make this configurable if we only accept `org.apache.kafka.common.serialization.StringSerializer` anyway?
Apart from that I am not sure if the name of the command line option will result in the desired outcome of putting key `key.serializer` to the Map. Have you tested if Spring Boot maps this as you expected?

Reply:
> why do we make this configurable if we only accept `org.apache.kafka.common.serialization.StringSerializer` anyway?

FMPOV this is not configurable. If you have an idea of how I can make it more clear in the description here, I'll be happy to change it. 
The logic is the following: every property that starts with the prefix `hono.kafka.producerConfig` is put into a map (without the prefix of course). The map is finally provided to the "native" Kafka client. You can add arbitrary properties there. The Kafka client applies the known properties and ignores those that it does not know. The table lists the exceptions: properties that are effectively not configurable because Hono overwrites them. You can set the command line option `--hono.kafka.producerConfig.key.serializer=bumlux`. Hono (i.e. the class `KafkaProducerConfigProperties`) will change the value of the property to `org.apache.kafka.common.serialization.StringSerializer` (a message about this is printed on debug level). 
I don't see a way to exclude properties from that mechanism. Alternatively, Hono would have to support each Kafka client config property explicitly (which are many dozens) and would then be tied to specific versions of the client (because properties change between versions).
Unless someone comes up with a better idea, I think simply overwriting the properties where we expect a certain value is an acceptable solution.

> Apart from that I am not sure if the name of the command line option will result in the desired outcome of putting key `key.serializer` to the Map. Have you tested if Spring Boot maps this as you expected?

I verified that it works as expected. Btw.: I was more surprised that Spring Boot applies its transformation magic (changing to lower case and replacing underscores by dots) not only on the level of the variable names (the prefix) but additionally with the part that it uses as keys in the map (therefore I thought initially we could not provide generic environment variables here). But it works fine.

Reply:
Ok, now I understand.
How about replacing
> The following table shows which properties _are_ changed by Hono.

and the table below with
> The following properties can *not* be set using this mechanism because the protocol adapters use fixed values instead in order to implement the message delivery semantics defined by Hono's Telemetry and Event APIs.

Then add a table with the kafka property names and the fixed values used by the adapters.